File: php/features.php
<?php
declare(strict_types=1);

// UselessFacts SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UselessFactsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UselessFactsBaseFeature();
            case "test":
                return new UselessFactsTestFeature();
            default:
                return new UselessFactsBaseFeature();
        }
    }
}

File: py/uselessfacts_sdk/features.py
# UselessFacts SDK feature factory

from uselessfacts_sdk.feature.base_feature import UselessFactsBaseFeature
from uselessfacts_sdk.feature.test_feature import UselessFactsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UselessFactsBaseFeature(),
        "test": lambda: UselessFactsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
